### ISSUE TITLE
Unpin typing_extensions

### ISSUE THREAD

**Issue by eivindjahren** [equinor/ert#5456]
https://github.com/equinor/ert/blob/59b26e5ec159909e695238e1e749d25ed4468398/setup.py#L128

ert_storage would fail with type_extensions==4.6 with

```
    @pytest.fixture
    def dark_storage_client(monkeypatch):
>       with dark_storage_app_(monkeypatch) as dark_app:

/home/runner/work/ert/ert/tests/unit_tests/dark_storage/conftest.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
/home/runner/work/ert/ert/tests/unit_tests/dark_storage/conftest.py:91: in dark_storage_app_
    from ert.dark_storage.app import app
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert/dark_storage/app.py:1: in <module>
    from ert_storage.app import JSONResponse
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/app.py:7: in <module>
    from ert_storage.endpoints import router as endpoints_router
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/endpoints/__init__.py:2: in <module>
    from .ensembles import router as ensembles_router
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/endpoints/ensembles.py:6: in <module>
    from ert_storage import database_schema as ds, json_schema as js
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/json_schema/__init__.py:3: in <module>
    from .experiment import ExperimentIn, ExperimentOut
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/json_schema/experiment.py:4: in <module>
    from .prior import Prior
/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/ert_storage/json_schema/prior.py:11: in <module>
    class PriorConst(BaseModel):
pydantic/main.py:197: in pydantic.main.ModelMetaclass.__new__
    ???
pydantic/fields.py:506: in pydantic.fields.ModelField.infer
    ???
pydantic/fields.py:436: in pydantic.fields.ModelField.__init__
    ???
pydantic/fields.py:552: in pydantic.fields.ModelField.prepare
    ???
pydantic/fields.py:668: in pydantic.fields.ModelField._type_analysis
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = typing.Tuple, cls = typing.Literal

    def __subclasscheck__(self, cls):
        if self._special:
            if not isinstance(cls, _GenericAlias):
>               return issubclass(cls, self.__origin__)
E               TypeError: issubclass() arg 1 must be a class

/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/typing.py:774: TypeError

```

**Comment by eivindjahren** [comment]
Blocked by https://github.com/pydantic/pydantic/issues/5821

**Comment by oysteoh** [comment]
Still the same error after upgrading `typing_extensions` to 4.7.1

**Comment by eivindjahren** [comment]
Closed by #5520
